content/5_Plat/2DRQ.mdx

```diff
@@ -259,58 +273,66 @@ easier to understand, albeit significantly slower due to a high constant factor:
 <CPPSection>
 
 ```cpp
-// index of largest value <= x in v (sorted)
-// if v = [1, 2, 4], ind(v, 3) would return 1
-int ind(vector<int> v, int x) {
-	return upper_bound(v.begin(), v.end(), x) - v.begin() - 1;
-}
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
 
-class BIT2D {
+/**
+ * Offline 2D Fenwick Tree implementation.
```

Comment:
what does offline mean

Reply:
it's explained in the module, basically u give it the location of the updates beforehand so u can do updates and queries in O(log^2(q)) or smth similar (and also rectangle sum queries do not need to be known beforehand)